README.md

```diff
@@ -332,6 +333,92 @@ to exist for the lifetime of the view controller object:
 
 ```
 
+### Error Handling ###
+
+The emergence of `try / catch` in Swift 2 has added powerful ways to define and return errors when something fails. The emergence of `ErrorType`
+as well for defining errors makes error definitions much more convenient over the cumbersome `NSError`. Because of this, for functions that can have multiple
+points of failure, you should always define it as `throws` and return a well-defined `ErrorType`.
+
+Consider the following contrived example:
+
+```swift
+
+func multiplyEvensLessThan10(evenNumber: Int) -> Int? {
+	guard evenNumber % 2 == 0 && evenNumber < 10 else {
+		return nil
+	}
+	
+	return evenNumber * 2
+}
+
+```
+
+The function above fails because it only expects evens less than 10 and returns an optional if that is violated. While this works and is simple, it
+is more Objective-C than Swift in its composition. The caller may not know which parameter they violated. For Swift, instead consider refactoring it as follows:
+
+```swift
+
+internal enum NumberError: ErrorType {
+	case NotEven
+	case TooLarge
+}
+
+func multiplyEvens(evenNumber: Int) throws -> Int {
+	guard evenNumber % 2 == 0 else {
+		throw NumberError.NotEven
+	}
+	
+	guard evenNumber < 10 else {
+		throw NumberError.TooLarge
+	}
+	
+	return evenNumber * 2
+}
+
+```
+
+The above, while slightly more cumbersome, this has two well-defined benefits:
+
+1. The caller is able to explicitly determine why their call to the function failed and thus can take active steps to recover:
+
+```swift
+    let result: Int
+    do {
+        result = try multiplyEvens(3)
+    } catch NumberError.NotEven {
+        return 0
+    } catch NumberError.TooLarge {
+        print("The Number entered was too large! Try again.")
+    } catch {
+        fatalError("Unhandled error occurred.")
+    }
+
+    return result
+```
+
+2. Try/catch semantics allow the caller to still retain the old optional functionality if the error is not relevant and they only care about the outcome:
```

Comment:
I updated it to use bullets instead; the numbering is weird with Markdown and I don't feel like fighting it.